### PR TITLE
Fix incorrect fields in docstrings

### PR DESCRIPTION
**Pull request by mthuurne**
## Contributor Checklist:

* [x] The associated ticket in Trac is here: https://twistedmatrix.com/trac/ticket/10021
* [x] I ran `tox -e black-reformat` to format my patch to meet the [Twisted Coding Standard](https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html)
* [x] I ran additional checks listed at [Getting Your Patch Accepted](https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted)
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: [News files](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles))
* [x] ~~I have updated the automated tests.~~
* [x] I have submitted the associated Trac ticket for review by adding the word `review` to the keywords field in Trac, and putting a link to this PR in the comment; it shows up in https://twisted.reviews/ now.

### CODE REVIEW

File: src/twisted/conch/ssh/keys.py [hunk 1]
```diff
@@ -1570,13 +1570,19 @@ def _toPrivateOpenSSH_PEM(self, passphrase=None):
...
+        @param subtype: A subtype to emit.  Only supported for private keys,
+            for which the currently supported subtypes are C{'PEM'} and C{'v1'}.
+            If not given, an appropriate default is used.
+        @type subtype: L{str} or L{None}
+
+        @param comment: Comment for a public key.
+        @type comment: L{bytes}
+
+        @param passphrase: Passphrase for a private key.
+        @type comment: L{bytes}
```
Comment (twm) [inline src/twisted/conch/ssh/keys.py]:
```suggestion
        @type passphrase: L{bytes}
```

File: src/twisted/internet/interfaces.py [hunk 2]
```diff
@@ -1588,9 +1588,6 @@ class IFileDescriptor(ILoggingContext):
 
     def fileno() -> object:
         """
-        @raise: If the descriptor no longer has a valid file descriptor
-            number associated with it.
```
Comment (twm) [inline src/twisted/internet/interfaces.py:1592]: It seems like a deficiency in the interface that there is no exception type defined for this. Oh well.

**Review by twm — APPROVED** [review]
This looks good. I didn't run pydoctor, but all these changes are clearly improvements. I did note one typo.

The only other thought I had is that there are several places documented as `Exception` that might technically produce `BaseException`. However that's always true so 🤷‍♂️ .

Thank you!

**Comment by mthuurne** [comment]
> The only other thought I had is that there are several places documented as `Exception` that might technically produce `BaseException`. However that's always true so man_shrugging .

I think the purpose of `@raise` is to document exceptions the caller might want to catch and the exceptions that inherit directly from `BaseException` are not ones you should be catching.

File: src/twisted/conch/ssh/keys.py (hunk 1 above)
Comment (mthuurne) [reply src/twisted/conch/ssh/keys.py]: Ah, that's indeed a copy-paste error on my part.

I guess that also answers the question of whether twisted/pydoctor#229 is worth implementing.

File: src/twisted/internet/interfaces.py (hunk 2 above)
Comment (mthuurne) [reply src/twisted/internet/interfaces.py:1592]: None of the implementations within Twisted raise an exception though. Some return -1 if the descriptor is no longer valid.

**Pull request merged**